rts/motoko-rts/src/region.rs

```diff
@@ -648,18 +674,23 @@ pub(crate) unsafe fn region_migration_from_v2plus_into_v2<M: Memory>(mem: &mut M
 //
 // region manager migration/initialization, with pre-existing stable data.
 //
-// Valid cases:
-//  - from version 1 into version 2.
-//  - from version 2 into version 2.
-//
 #[ic_mem_fn]
-pub(crate) unsafe fn region_init<M: Memory>(mem: &mut M, from_version: i32) {
+pub(crate) unsafe fn region_init<M: Memory>(mem: &mut M, use_stable_regions: i32) {
     debug_assert!(meta_data::offset::FREE < meta_data::offset::BLOCK_ZERO);
-    match from_version {
-        0 => region_migration_from_v0_into_v2(mem),
-        1 => region_migration_from_v1_into_v2(mem),
-        2 => region_migration_from_v2plus_into_v2(mem),
-        _ => unreachable!(),
+    match crate::stable_mem::get_version() {
+        VERSION_NO_STABLE_MEMORY => {
+            assert!(crate::stable_mem::size() == 0);
+            if use_stable_regions != 0 {
+                region_migration_from_no_stable_memory(mem);
+            };
+        }
+        VERSION_SOME_STABLE_MEMORY => {
+            assert!(crate::stable_mem::size() > 0);
+            if use_stable_regions != 0 {
+                region_migration_from_some_stable_memory(mem);
+            };
+        }
+        _ => region_migration_from_regions_plus(mem), //check format & recover region0
```

Comment:
Possible assertion: `use_stable_region` would be `true` (`!=0`) in this case.

Reply:
The flag now means for migration to regions after upgrade, so (I think) it may not be true in this case. I'm mostly keeping the flag to avoid rewriting all the tests, but we could jettison it.